src/charts/donut/donut-pct-chart-directive.js

```diff
@@ -0,0 +1,354 @@
+     };
+
+     $scope.newUsed = $scope.data.used;
+
+     $scope.$watch('newUsed', function(val) {
+       $scope.data.used = val;
+     });
+
+     $scope.usedConfig = {
+       'units': 'GB',
+       'thresholds':{'warning':'60','error':'90'}
+     };
+
+     $scope.usedData = {
+         'used': '350',
+         'total': '1000'
+     };
+     $scope.availConfig = {
+       'units': 'GB',
+       'thresholds':{'warning':'60','error':'90'}
+     };
+
+     $scope.availData = {
+         'used': '350',
+         'total': '1000'
+     };
+     $scope.pctConfig = {
+       'units': 'GB',
+       'thresholds':{'warning':'60','error':'90'}
+     };
+
+     $scope.pctData = {
+         'used': '350',
+         'total': '1000'
+     };
+     $scope.noneConfig = {
+       'units': 'GB',
+       'thresholds':{'warning':'60','error':'90'}
+     };
+
+     $scope.noneData = {
+         'used': '350',
+         'total': '1000'
+     };
+
+     $scope.custConfig = {
+       'units': 'MHz',
+       'thresholds':{'warning':'60','error':'90'},
+       "legend":{"show":true},
+       'tooltipFn': function (d) {
+         return '<span class="donut-tooltip-pf"style="white-space: nowrap;">' +
+                   d[0].value + ' ' + d[0].name +
+                '</span>';
+       },
+       'centerLabelFn': function (scope) {
+         return '<tspan dy="0" x="0" class="donut-title-big-pf">' +
+                  scope.data.available + '</tspan>' +
+                '<tspan dy="20" x="0" class="donut-title-small-pf">' +
+                  'Free' +
+                '</tspan>';
+       }
+     };
+
+     $scope.custData = {
+         'used': '670',
+         'total': '1000'
+     };
+
+   };
+ </file>
+ </example>
+ */
+angular.module('patternfly.charts').directive('pfDonutPctChart', ['ChartsMixin', '$timeout',
+  function(chartsMixin, $timeout) {
+    'use strict';
+    return {
+      restrict: 'A',
+      scope: {
+        id: '@',
+        config: '=',
+        data: '=',
+        centerLabel: '@'
+      },
+      replace: true,
+      templateUrl: 'charts/donut/donut-pct-chart.html',
+      controller: ['$scope',
+        function($scope) {
+
+          $scope.donutChartId = 'donutChart';
+          if ($scope.id) {
+            $scope.donutChartId = $scope.id + $scope.donutChartId;
+          }
+
+          $scope.updateAvailable = function(){
+            $scope.data.available = $scope.data.total - $scope.data.used;
+          };
+
+          if ($scope.data.available === undefined) {
+            $scope.updateAvailable();
+          }
+
+          $scope.getStatusColor = function(used, thresholds) {
+            if (!thresholds){
+              return '#0088CE';
+            } else {
+              if (used >= thresholds.error) {
+                return '#CC0000';
+              } else if (used >= thresholds.warning) {
+                return '#EC7A08';
+              } else {
+                return '#0088CE';
+              }
+            }
```

Comment:
This logic is very confusing to me because of the negative condition and the multiple return points.  I suggest reworking it like this (untested):

```javascript
$scope.getStatusColor = function(used, thresholds) {
    var color = '#0088CE';

    if (thresholds) {
        if (used >= thresholds.error) {
            color = '#CC0000';
        } else if (used >= thresholds.warning) {
            color = '#EC7A08';
        }
    }
 
    return color;
};
```